internal/nginx/modules/httpmatches.js

```diff
@@ -0,0 +1,94 @@
+export default { redirect };
+
+const matchesVariable = 'http_matches';
+
+// FIXME(osborn): Need to add special handling for repeated headers.
+// Should follow guidance from https://www.rfc-editor.org/rfc/rfc7230.html#section-3.2.2.
+function headersMatch(r, headers) {
+  for (let i = 0; i < headers.length; i++) {
+    const h = headers[i];
+    const kv = h.split(':');
+    if (kv.length !== 2) {
+      r.error('invalid header match: ' + h);
+      throw 500;
+    }
+    let val = r.headersIn[kv[0]];
+    if (!val || val.toLowerCase() !== kv[1]) {
```

Comment:
I don't think we can lower case here, for instance, if the value is base64 encoded "AAA" does not equal "aaa".

`X_BASE64_HEADER=Y2FzZSBzZW5zaXRpdmU=` // "case sensitive"
`X_BASE64_HEADER=y2fzzsbzzw5zaxrpdmu=` // "gskvk"